src/App.tsx

```diff
@@ -15,36 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum ESortType {
+  ALPHABETICALLY = 'alphabetically',
+  LENGTH = 'by length',
+  DEFAULT = '',
+}
+
+interface IOptions {
+  sortBy: ESortType;
+  isReversed: boolean;
+}
+
+const getGoods = (goods: string[], { sortBy, isReversed }: IOptions) => {
+  let visibleGoods = [...goods];
+
+  if (isReversed) {
+    visibleGoods = visibleGoods.toReversed();
+  }
+
+  if (sortBy) {
+    switch (sortBy) {
+      case ESortType.ALPHABETICALLY:
+        visibleGoods = visibleGoods.toSorted();
+        if (isReversed) {
+          visibleGoods = visibleGoods.toReversed();
+        }
+
+        break;
+      case ESortType.LENGTH:
+        if (isReversed) {
+          visibleGoods = visibleGoods.toSorted((a, b) => b.length - a.length);
+        } else {
+          visibleGoods = visibleGoods.toSorted((a, b) => a.length - b.length);
+        }
+
+        break;
+      default:
+        return null;
+    }
+  }
+
+  return visibleGoods;
+};
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState<ESortType>(ESortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+  const canReset = sortBy || isReversed;
+
+  const visibleGoods = getGoods(goodsFromServer, { sortBy, isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortBy !== ESortType.ALPHABETICALLY,
+          })}
+          onClick={() => setSortBy(ESortType.ALPHABETICALLY)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button is-success', {
+            'is-light': sortBy !== ESortType.LENGTH,
+          })}
+          onClick={() => setSortBy(ESortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {canReset ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortBy(ESortType.DEFAULT);
+              setIsReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        ) : null}
```

Comment:
we can use `&&` operator, it will work the same and there will be no need in empty ternary condition
```suggestion
        {canReset && (
          <button
            type="button"
            className="button is-danger is-light"
            onClick={() => {
              setSortBy(ESortType.DEFAULT);
              setIsReversed(false);
            }}
          >
            Reset
          </button>
        )}
```